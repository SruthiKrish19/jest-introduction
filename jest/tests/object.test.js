const { ageIsNumber, hasContactDetails } = require("../object");
const { validDetails, invalidDetails } = require("./test-data");

describe("ageIsNumber function", () => {
  // Positive Test Case
  test("returns true for all valid ages", () => {
    expect(ageIsNumber(validDetails)).toEqual([true, true]);
  });
  // Negative Test Case
  test("returns false if age is not a number", () => {
    expect(ageIsNumber(invalidDetails)).toEqual([false, false]);
  });
});

describe("hasContactDetails function", () => {
  // Positive Test Case
  test("returns true if person has phone or email", () => {
    expect(hasContactDetails(validDetails)).toEqual([true, true]);
  });
  // Negative Test Case
  test("returns false if person has no phone or email", () => {
    expect(hasContactDetails(invalidDetails)).toEqual([false, false]);
  });
});

// Snapshots example
test("snapshot of valid person details", () => {
  expect(validDetails).toMatchSnapshot();
});

test("snapshot of invalid person details", () => {
  expect(invalidDetails).toMatchSnapshot();
});
