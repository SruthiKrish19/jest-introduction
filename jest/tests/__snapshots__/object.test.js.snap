// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of invalid person details 1`] = `
[
  {
    "age": "thirty-two",
    "email": "",
    "gender": "male",
    "name": "John Doe",
    "phone": "",
  },
  {
    "age": "thirty-one",
    "email": "",
    "gender": "female",
    "name": "Jane Doe",
    "phone": "",
  },
]
`;

exports[`snapshot of valid person details 1`] = `
[
  {
    "age": 32,
    "email": "john@xyz.com",
    "gender": "male",
    "name": "John Doe",
    "phone": "1234567890",
  },
  {
    "age": 31,
    "email": "jane@xyz.com",
    "gender": "female",
    "name": "Jane Doe",
    "phone": "1234567899",
  },
]
`;
